Add AOTI model name config

yushangdi opened the pull request:
Summary: If a model name is specified in aoti config, the generated files will use that model name as file stem.

Test Plan:
```
buck2 run mode/dev-nosan caffe2/test/inductor:test_aot_inductor -- -r test_using_model_name_for_files
```

Bifferential Revision: D75102034


cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

desertfire commented on torch/_inductor/codecache.py:
```diff
@@ -1638,6 +1641,7 @@ def compile(
             "wrapper.cpp",
             extra=cpp_command,
             specified_dir=specified_output_path,
+            key=model_name,
```
Can we just use `config.aot_inductor.model_name_for_generated_file` here? Then we won't need to update `compile_fx.py`.

yushangdi replied:
@desertfire Fixed!

desertfire commented:
Need to fix CI issues

yushangdi commented:
> Need to fix CI issues

@desertfire  fixed now

desertfire reviewed (approved):
Looks like there is a test conflict to be resolved

yushangdi committed:
Add AOTI model name config (#154129)

Summary:

If a model name is specified in aoti config, the generated files will use that model name as file stem.

Test Plan:
```
buck2 run mode/dev-nosan caffe2/test/inductor:test_aot_inductor -- -r test_using_model_name_for_files
```

Reviewed By: desertfire, muchulee8, ColinPeppler

Differential Revision: D75102034

jbschlosser commented on torch/_inductor/config.py:
```diff
@@ -1346,6 +1350,10 @@ class aot_inductor:
     # In addition to emit asm files, also emit binary files for current arch
     emit_current_arch_binary: bool = False
 
+    # If not None, the generated files with use this name in file stem.
```
```suggestion
    # If not None, the generated files will use this name in file stem.
```

seemethere commented:
@pytorchbot merge -f 'merged internally'
